src/execution_plan/optimizations/cost_base_label_scan.c

```
@@ -34,6 +34,231 @@
+	// X, Y & Z are destination labels
+
+	uint n;
+	AlgebraicExpression **operands =
+		AlgebraicExpression_CollectOperandsInOrder(ae, &n);
+
+	uint src_ops_n  = 0;               // number of source labels
+	uint dest_ops_n = 0;               // number of destination labels
+	AlgebraicExpression *src_ops[n];   // source labels
+	AlgebraicExpression *dest_ops[n];  // destination labels
+
+	// populate source labels array
```

Cppcheck (2.10-2)

[error] Common realloc mistake

(memleakOnRealloc)